lib/receipt.js

```diff
@@ -229,7 +230,7 @@ function hashReceipt(serializedReceipt) {
     // Due to how Solidity encodes and hashes uint8 values, we must explicitly
     // hash them as uint8 values to avoid Solidity encoding and hashing them
     // as uint256 values
-    const receiptCopy = JSON.parse(JSON.stringify(serializedReceipt));
+    const receiptCopy = cloneDeep(serializedReceipt);
```

Comment:
I would  really like to see a unit test demonstrating that this is need, otherwise it might just get reverted by unintentional cleanups!

Reply:
Actually, after I converted all the bignumber type variables into strings for the payment receipt fields in the test, this change is not necessary. Thanks for reminding me to do further checks.